### PR TITLE
 feat(session): add component navbar

### PR DESCRIPTION
**Pull request by YgRamos9**
## Summary
- adding component auth

## Screenshot
Captura de Pantalla 2022-08-07 a la(s) 3 56 57 p m

## Trello Card
https://trello.com/c/VtYKfim0/9-session-management-logout

### CODE REVIEW

**Comment by YgRamos9** [comment]
## Summary
- adding component auth

## Screenshot
Captura de Pantalla 2022-08-07 a la(s) 3 56 57 p m

## Trello Card
https://trello.com/c/VtYKfim0/9-session-management-logout

File: angular/src/app/app.component.scss [hunk 1]
```diff
@@ -3,4 +3,5 @@
   display: flex;
   height: 100vh;
   justify-content: center;
+
```
Comment (joacomiguens) [inline angular/src/app/app.component.scss]: delete 😄

File: angular/src/app/screens/auth/auth.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { CommonModule } from '@angular/common';
+import { HttpClientModule } from '@angular/common/http';
+import { render } from '@testing-library/angular';
+import { screen } from '@testing-library/dom';
+import '@testing-library/jest-dom';
+import { NavbarComponent } from '../../components/navbar/navbar.component';
+import { AuthRoutingModule } from './auth-routing.module';
+import { AuthComponent } from './auth.component';
+
+describe('Render auth component', () => {
```
Comment (joacomiguens) [inline angular/src/app/screens/auth/auth.component.spec.ts:11]: I think you could test what happens when user logout

File: angular/src/app/screens/login/login.component.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ import { StatusRequest } from '../../constants/code-request';
 import { LOGIN_FIELDS } from '../../constants/form-account';
 import { IUser } from 'src/app/interfaces/global.interface';
 import { UserService } from '../../services/user.service';
+import { Router } from '@angular/router';
```
Comment (joacomiguens) [inline angular/src/app/screens/login/login.component.ts]: must be at top

File: angular/src/app/screens/login/login.component.ts [hunk 4]
```diff
@@ -13,9 +14,10 @@ import { FormControl, FormGroup, Validators } from '@angular/forms';
 export class LoginComponent {
   fields = LOGIN_FIELDS;
   buttonSend = 'Login';
+  mensajeError!: string;
```
Comment (joacomiguens) [inline angular/src/app/screens/login/login.component.ts]: why? 🤔

File: angular/src/app/screens/login/login.component.ts [hunk 5]
```diff
@@ -30,7 +32,11 @@ export class LoginComponent {
     this.userService.login(value).subscribe(
       (response: HttpResponse<Object>) => {
         if (response.status === StatusRequest.Ok) {
-          console.log(response.headers.get('Access-Token'));
+          const token = response.headers.get('Access-Token');
+          if (token) {
+            this.userService.createSession(token);
```
Comment (joacomiguens) [inline angular/src/app/screens/login/login.component.ts]: You need to save access-token, uid, client. Storage it as an unique string in a single slot on localStorage 👌🏻

File: angular/src/app/services/user.service.ts [hunk 6]
```diff
@@ -33,4 +33,16 @@ export class UserService {
       },
     );
   }
+
+  createSession(token: string): void {
+    localStorage.setItem('userToken', token);
+  }
+
+  signOff() {
```
Comment (joacomiguens) [inline angular/src/app/services/user.service.ts]:
```suggestion
  logOut() {
```

File: angular/src/app/services/user.service.ts [hunk 7]
```diff
@@ -33,4 +33,16 @@ export class UserService {
...
+    localStorage.removeItem('userToken');
+  }
+
+  isLogin() {
```
Comment (joacomiguens) [inline angular/src/app/services/user.service.ts]: Do you use this function? 🖲️

File: angular/src/app/components/navbar/navbar.component.ts [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+import { UserService } from '../../services/user.service';
+
+@Component({
+  selector: 'app-navbar',
+  templateUrl: './navbar.component.html',
+  styleUrls: ['./navbar.component.scss'],
+})
+export class NavbarComponent {
+  constructor(private userService: UserService, private router: Router) {}
+  logOut() {
```
Comment (miguelhenaoa) [inline angular/src/app/components/navbar/navbar.component.ts:13]: blank line

File: angular/src/app/services/user.service.ts [hunk 9]
```diff
@@ -33,4 +33,16 @@ export class UserService {
       },
     );
   }
+
+  createSession(token: string): void {
```
Comment (miguelhenaoa) [inline angular/src/app/services/user.service.ts]:
```suggestion
  setToken(token: string): void {
```

**Review by miguelhenaoa — CHANGES_REQUESTED** [review]

**Review by joacomiguens — APPROVED** [review]

File: angular/src/app/screens/auth/auth.component.spec.ts (hunk 2 above)
Comment (joacomiguens) [reply angular/src/app/screens/auth/auth.component.spec.ts:11]: Do it on guards PR

**Review by miguelhenaoa — APPROVED** [review]

**Pull request merged**